alpha-commons/src/main/java/at/ac/tuwien/kr/alpha/commons/PredicateImpl.java

```diff
@@ -1,33 +1,39 @@
-package at.ac.tuwien.kr.alpha.common;
+package at.ac.tuwien.kr.alpha.commons;
+
+import at.ac.tuwien.kr.alpha.api.programs.Predicate;
+import at.ac.tuwien.kr.alpha.commons.util.Interner;
 
 /**
- * Copyright (c) 2016, the Alpha Team.
+ * A predicate as used by the Alpha solver internally.
+ * 
+ * Copyright (c) 2016-2020, the Alpha Team.
  */
-public class Predicate implements Comparable<Predicate> {
-	private static final Interner<Predicate> INTERNER = new Interner<>();
+class PredicateImpl implements Predicate {
+	
+	private static final Interner<PredicateImpl> INTERNER = new Interner<>();
 
 	private final String name;
 	private final int arity;
 	private final boolean internal;
 	private final boolean solverInternal;
 
-	protected Predicate(String name, int arity, boolean internal, boolean solverInternal) {
+	PredicateImpl(String name, int arity, boolean internal, boolean solverInternal) {
 		this.name = name;
 		this.arity = arity;
 		this.internal = internal;
 		this.solverInternal = solverInternal;
 	}
 
-	public static Predicate getInstance(String symbol, int arity) {
+	static PredicateImpl getInstance(String symbol, int arity) {
 		return getInstance(symbol, arity, false, false);
 	}
```

Comment:
This method is never used as `Predicates` does its job already.
```suggestion
```

Reply:
Deleted in d7f71891